src/services/university.service.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  DonationsByMonth,
+  DonationsByParty,
+  University,
+  TopRecipientsDollar,
+  TopRecipientsDonation,
+  IdeologyDistribution,
+  TotalContributionsDollar,
+  RegisteredVoters,
+} from '@interfaces/university.interface';
+import { HttpException } from '@exceptions/HttpException';
+import { Prisma } from '@prisma/client';
+import prismaClient from '@databases/client';
+
+class UniversityService {
+  public async getUniversityData(uniId: string, startDate: string, endDate: string): Promise<any> {
+    // First check if uni exists (fetch uniinfo such as name, industry)
+    console.log('here');
+    const uniInfo: University = await prismaClient.organization.findUnique({
+      where: {
+        id: uniId,
+      },
+    });
+    if (uniInfo === null || uniInfo.industry !== 'school') {
+      throw new HttpException(404, 'University not found.');
+    }
+
+    // Setup date objects to be used for query (start/end date for the requested period)
+    const start_date = new Date(startDate);
+    const end_date = new Date(endDate);
+
+    // Then, proceed with queries
+    const [
+      donationsByMonth,
+      donationsByParty,
+      topRecipientsDollar,
+      topRecipientsDonation,
+      ideologyDistribution,
+      totalContributionsDollar,
+      registeredVoters,
+    ]: [
+      DonationsByMonth,
+      DonationsByParty,
+      TopRecipientsDollar,
+      TopRecipientsDonation,
+      IdeologyDistribution,
+      TotalContributionsDollar,
+      RegisteredVoters,
+    ] = [
+      // Note: Prisma's groupBy function is broken.
+      // Donations across time (grouped by month)
+      await prismaClient.$queryRaw<DonationsByMonth>(
+        Prisma.sql`
+          SELECT
+            DATE_TRUNC('month',date) AS month_start_date,
+            SUM(amount)::float AS amount_donated
+          FROM donation
+          WHERE org_id = ${uniId}
+          GROUP BY DATE_TRUNC('month',date)
+          ORDER BY month_start_date;`,
+      ),
+      // DonationsByParty
+      await prismaClient.$queryRaw<DonationsByParty>(Prisma.sql`
+          SELECT
+            party,
+            SUM(amount)::float as total_amount
+          FROM donation as d
+          JOIN recipient as r
+            ON d.rec_id = r.id
+          WHERE org_id = ${uniId} AND d.date BETWEEN ${start_date} AND ${end_date}
+          GROUP BY party
+          ORDER BY SUM(amount) DESC
+          LIMIT 3;`),
```

Comment:
Just a note that we should put these arbitrary numbers somewhere later, include the ones I used for the queries I added